demos/llamastack/Llama_Stack_NVIDIA_E2E_Flow.ipynb

```diff
@@ -0,0 +1,2924 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "uv sync --extra dev\n",
+    "uv pip install -U llama-stack-client\n",
+    "uv pip install -e .\n",
+    "source .venv/bin/activate\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Build Llama Stack Image\n",
+    "Build the Llama Stack image using the virtual environment you just created. For local development, set `LLAMA_STACK_DIR` to ensure your local code is use in the image. To use the production version of `llama-stack`, omit `LLAMA_STACK_DIR`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```bash\n",
+    "uv run --with llama-stack llama stack list-deps nvidia | xargs -L1 uv pip install\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "1. Update the following variables in [config.py](./config.py) with your deployment URLs and API keys. The other variables are optional. You can update these to organize the resources created by this notebook.\n",
+    "```python\n",
+    "# (Required) NeMo Microservices URLs\n",
+    "NDS_URL = \"\" # NeMo Data Store\n",
+    "NEMO_URL = \"\" # Other NeMo Microservices (Customizer, Evaluator, Guardrails)\n",
+    "NIM_URL = \"\" # NIM\n",
+    "\n",
+    "# (Required) Hugging Face Token\n",
+    "HF_TOKEN = \"\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2. Set environment variables used by each service."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from config import *\n",
+    "\n",
+    "# Metadata associated with Datasets and Customization Jobs\n",
+    "os.environ[\"NVIDIA_DATASET_NAMESPACE\"] = NAMESPACE\n",
+    "os.environ[\"NVIDIA_PROJECT_ID\"] = PROJECT_ID\n",
+    "\n",
+    "# Inference env vars\n",
+    "os.environ[\"NVIDIA_BASE_URL\"] = NIM_URL\n",
+    "\n",
+    "# Data Store env vars\n",
+    "os.environ[\"NVIDIA_DATASETS_URL\"] = NEMO_URL\n",
+    "\n",
+    "# Customizer env vars\n",
+    "os.environ[\"NVIDIA_CUSTOMIZER_URL\"] = NEMO_URL\n",
+    "os.environ[\"NVIDIA_OUTPUT_MODEL_DIR\"] = CUSTOMIZED_MODEL_DIR\n",
+    "\n",
+    "# Evaluator env vars\n",
+    "os.environ[\"NVIDIA_EVALUATOR_URL\"] = NEMO_URL\n",
+    "\n",
+    "# Guardrails env vars\n",
+    "os.environ[\"GUARDRAILS_SERVICE_URL\"] = NEMO_URL\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "3. Initialize the HuggingFace API client. Here, we use NeMo Data Store as the endpoint the client will invoke."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
```

Comment:
I don't think it is necessary to keep your outputs in the notebook. 
There are a few other places where outputs are kept in this notebook. I will not add comments to all of them.